ethereum/spec/src/main/java/tech/pegasys/teku/spec/ForkManifest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.spec;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import tech.pegasys.teku.datastructures.state.Fork;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.constants.SpecConstants;
+import tech.pegasys.teku.ssz.SSZTypes.Bytes4;
+
+public class ForkManifest {
+  private final List<Fork> forkSchedule;
+
+  public ForkManifest(SpecConstants genesisConstants) {
+    forkSchedule = new ArrayList<>();
+    final Bytes4 genesisForkVersion = genesisConstants.getGenesisForkVersion();
+    final UInt64 genesisEpoch = UInt64.valueOf(genesisConstants.getGenesisEpoch());
+    forkSchedule.add(new Fork(genesisForkVersion, genesisForkVersion, genesisEpoch));
+  }
+
+  @VisibleForTesting
+  void addFork(final Fork fork) {
+    final Fork lastFork = forkSchedule.get(forkSchedule.size() - 1);
+    checkArgument(fork.getEpoch().isGreaterThan(lastFork.getEpoch()));
+    forkSchedule.add(fork);
+  }
```

Comment:
Would be nice to make `ForkManifest` immutable.  A few options for doing that:
* Make the constructor private, accepting a list of `Fork`'s, add some static factory methods:
  * `create(SpecConstants genesisConstants)`
  * `create(List<Fork>)`
* Create a `Fork.Builder` class